Implement poolingLayer using WebNN API and add more detailed logs

MarkGHX opened the pull request:
@huningxin Hi Ningxin, this PR includes the implementation of poolingLayer using WebNN API and some detailed logs. You can use the following codes to check it:
```shell
$ ./bin/opencv_test_dnn --gtest_filter=*Test_ONNX_layers.MaxPooling/0*
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from Test_ONNX_layers
[ RUN      ] Test_ONNX_layers.MaxPooling/0, where GetParam() = WEBNN/CPU
[ WARN:0] global /home/webml/GSoC2021/opencv/modules/dnn/src/layers/pooling_layer.cpp (266) supportBackend ceilMode is not supported by WebNN backend.
[ WARN:0] global /home/webml/GSoC2021/opencv/modules/dnn/src/dnn.cpp (2460) initWebnnBackend Layer Pooling name 1 is unsupported by WebNN backend.
[       OK ] Test_ONNX_layers.MaxPooling/0 (2 ms)
[----------] 1 test from Test_ONNX_layers (2 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (2 ms total)
[  PASSED  ] 1 test.
```
Thanks a lot!

huningxin commented:
@MarkGHX , let's move the discussion to https://github.com/opencv/opencv/pull/20406. So OpenCV guys can also watch and comment. Please also add the test results there.

MarkGHX commented:
OK, I will repost the test log in opencv#20406.

sturkmen72 committed:
Update perf_bgfg_mog2.cpp, perf_bgfg_knn.cpp

alalek committed:
Merge pull request #20748 from sturkmen72:clean_up_test_data

sturkmen72 committed:
Update OpenCVDetectVTK.cmake

smirnov-alexey committed:
Add more parameters to render fixture

alalek committed:
Merge pull request #20763 from Cavendish-Koo:houghlines_sdiv_bug_fix

shengyu7697 committed:
remove redundant semicolons

alalek committed:
dnn(ocl): fix conv BASIC workgroup

alalek committed:
Merge pull request #20776 from alalek:port_20748_4.x

alalek committed:
Merge pull request #20774 from alalek:dnn_ocl_fix_conv_basic_workgroup

alalek committed:
Merge pull request #20764 from sturkmen72:Update_OpenCVDetectVTK_cmake

TolyaTalamanov committed:
Fix bugs with hanging frames

sturkmen72 committed:
restore LSD

tomoaki0705 committed:
suppress GaussianBlur to generate empty images
  * sharp Gaussian kernel causes over flow and ends up in blank image

alalek committed:
Merge pull request #20791 from TolyaTalamanov:at/dangling-media-frames

alalek committed:
Merge pull request #20793 from tomoaki0705:fixGaussianOverflow

alalek committed:
dnn(ocl): fix conv DWCONV workgroup

JBamberger committed:
Automatically set the correct OpenCV version in build.gradle

Automatically sets the correct OpenCV version in the CMAKE example contained in the build.gradle
file of the Android SDK.

alalek committed:
Merge pull request #20794 from alalek:dnn_ocl_fix_conv_dwconv_workgroup

JonasVautherin committed:
Fix gst_initializer

Use the return value of gst_init_check instead of testing the error pointer

sturkmen72 committed:
additional changes

alalek committed:
Merge pull request #20795 from JBamberger:gradle-dynamic-cv-version

alalek committed:
Merge pull request #20796 from JonasVautherin:patch-1

alalek committed:
Merge pull request #20797 from sturkmen72:restore_LSD

alalek committed:
Merge remote-tracking branch 'upstream/3.4' into merge-3.4

jebastin-nadar committed:
fix illegal memory access in int8 convolution

alalek committed:
Merge pull request #20800 from SamFC10:fix-int8-conv

YashasSamaga committed:
support broadcasting in eltwise ops

alalek committed:
core(TLS): force TlsAbstraction initialization before main()

alalek committed:
core(parallel): suppress TSAN warning

sthalik committed:
fix OPENCV_DISABLE_THREAD_SUPPORT

Message was,
  CMake Error at CMakeLists.txt:1475 (message):
    Not all parallel frameworks have been disabled (using Concurrency).

sthalik committed:
add new supported MSVC version

sthalik committed:
fix link error on shared libs with -MT

alalek committed:
Merge pull request #20772 from shengyu7697:remove-redundant-semicolon

alalek committed:
valgrind: update suppressions

zihaomu committed:
Merge pull request #20702 from zihaomu:tf_expand_dim_layer

Add ExpandDims layer of tf_importer.cpp

* Add ExpandDims to tf_importer.

* add -1 expand test case.

* Support different dimensions of input.

* Compatible with 5-dimensional NDHWC data

* Code align

* support 3-dim input.

* 3-dim bug fixed.

* fixing error of code format.

xhawk18 committed:
Merge pull request #20183 from xhawk18:3.4

* improve compatibility for qt 6.

* cmake(highgui): rework Qt dependency support

* cmake(highgui): workaround Qt5Config.cmake "components" bug

Co-authored-by: Alexander Alekhin <alexander.a.alekhin@gmail.com>

keroiber committed:
Merge pull request #20743 from keroiber:prefix_js_function_bindings_with_namespace

* Prefix global javascript functions with sub-namespaces

* js: handle 'namespace_prefix_override', update filtering

- avoid functions override with same name but different namespace

Co-authored-by: Alexander Alekhin <alexander.a.alekhin@gmail.com>

jebastin-nadar committed:
Merge pull request #20535 from SamFC10:onnx-q

dnn : int8 quantized layers support in onnx importer

* added quantized layers support in onnx importer

* added more cases in eltwise node, some more checks

* added tests for quantized nodes

* relax thresholds for failed tests, address review comments

* refactoring based on review comments

* added support for unsupported cases and pre-quantized resnet50 test

* relax thresholds due to int8 resize layer

spazewalker committed:
Merge pull request #20291 from spazewalker:master

speech recognition sample

* speech recognition sample added.(initial commit)

* fixed typos, removed plt

* trailing whitespaces removed

* masking removed and using opencv for displaying spectrogram

* description added

* requested changes and add opencl fp16 target

* parenthesis and halide removed

* workaround 3d matrix issue

* handle multi channel audio

support for multiple files at once

* suggested changes

fix whitespaces

alalek committed:
Merge pull request #20803 from alalek:core_parallel_workaround_tsan

alalek committed:
Merge remote-tracking branch 'upstream/3.4' into merge-3.4

alalek committed:
pre: OpenCV 3.4.16 (version++)

alalek committed:
Merge pull request #20782 from YashasSamaga:cuda4dnn-eltwise-broadcast

alalek committed:
pre: OpenCV 4.5.4 (version++)

alalek committed:
Merge pull request #20806 from alalek:update_version_3.4.16-pre

alalek committed:
ffmpeg/3.4: update FFmpeg wrapper 2021.10

- FFmpeg 3.4.8 (no changes)

alalek committed:
Merge pull request #20788 from alalek:update_version_4.5.4-pre

komakai committed:
Simple matrix multiplication for Mat in iOS/Android

alalek committed:
ffmpeg/4.x: update FFmpeg wrapper 2021.10

- FFmpeg 4.4 (no changes)

alalek committed:
Merge pull request #20771 from smirnov-alexey:as/render_fixture_params

rogday committed:
add SoftNMS implementation

alalek committed:
Merge pull request #20808 from alalek:update_ffmpeg_3.4

alalek committed:
Merge pull request #20811 from alalek:update_ffmpeg_4.x

hanliutong committed:
Merge pull request #20521 from hanliutong:dev-rvv-multiVLEN

Make the implementation of optimization in DNN adjustable to different vector sizes with RVV intrinsics.

* Update fastGEMM for multi VLEN.

* Update fastGEMM1T for multi VLEN.

* Update fastDepthwiseConv for multi VLEN.

* Update fastConv for multi VLEN.

* Replace malloc with cv::AutoBuffer.

alalek committed:
Merge pull request #20731 from komakai:matrix_mult_android_ios

alalek committed:
dnn(pytest/test_input_3d): reload model between switching targets

alalek committed:
Merge pull request #20802 from alalek:core_tls_init_TlsAbstraction

alalek committed:
Merge pull request #20814 from alalek:dnn_pytest_input_3d

alalek committed:
Merge pull request #20815 from alalek:fix_20649_revert_19859

features2d: repair SimpleBlobDetector

* features2d: revert code change by PR #19859

Reverted commit 76860933f0a31c0abd1b26c1f11b25972cda031e

* features2d: check SimpleBlobDetector parameters consistency

komakai committed:
Fix build on older Xcode versions

rogday committed:
backport YOLOv4x-mish new_coords CUDA implementation

mologie committed:
Merge pull request #20725 from mologie:fix-dnn-tf-on-arm

* dnn: fix unaligned memory access crash on armv7

The getTensorContent function would return a Mat pointing to some
member of a Protobuf-encoded message. Protobuf does not make any
alignment guarantees, which results in a crash on armv7 when loading
models while bit 2 is set in /proc/cpu/alignment (or the relevant
kernel feature for alignment compatibility is disabled). Any read
attempt from the previously unaligned data member would send SIGBUS.

As workaround, this commit makes an aligned copy via existing clone
functionality in getTensorContent. The unsafe copy=false option is
removed. Unfortunately, a rather crude hack in PReLUSubgraph in fact
writes(!) to the Protobuf message. We limit ourselves to fixing the
alignment issues in this commit, and add getTensorContentRefUnaligned
to cover the write case with a safe memcpy. A FIXME marks the issue.

* dnn: reduce amount of .clone() calls

* dnn: update FIXME comment

Co-authored-by: Alexander Alekhin <alexander.a.alekhin@gmail.com>

alalek committed:
android: drop sourceCompatibility/targetCompatibility options

alalek committed:
videoio: fix plugins handling if no filesystem available

alalek committed:
Merge pull request #20817 from komakai:fix_old_xcode_build

alalek committed:
core: ensure is_trivially_copyable for simple types

alalek committed:
highgui: repair Qt backend

alalek committed:
Merge pull request #20813 from rogday:soft_nms

alalek committed:
Merge pull request #20826 from alalek:fix_20824

alalek committed:
dnn(ocl): skip int8 tests due to memory access issues

alalek committed:
Merge pull request #20829 from alalek:dnn_ocl_skip_int8_tests

alalek committed:
Merge pull request #20819 from alalek:issue_19567

alalek committed:
Merge pull request #20820 from alalek:issue_20514

alalek committed:
Merge pull request #20825 from alalek:issue_20718

alalek committed:
Merge remote-tracking branch 'upstream/3.4' into merge-3.4

alalek committed:
core: fix compilation of copy ctors/assignment operators with GCC 4.x

alalek committed:
Merge pull request #20831 from sthalik:fix-msvc-build-3.4

alalek committed:
Merge pull request #20835 from sthalik:msvc-no-thread-support

alalek committed:
dnn(ocl4dnn): add extra checks to convolution layer

- prevent running code over unsupported/non-tested configurations
- prevent integer div by zero

Commit:
Merge pull request #20823 from AleksandrPanov:fix_orb_integer_overflow

Fix ORB integer overflow

* set size_t step to fix integer overflow in ptr0 offset

* added issue_537 test

* minor fix tags, points

* added size_t_step and offset to remove mixed unsigned and signed operations

* features2d: update ORB checks

Co-authored-by: Alexander Alekhin <alexander.a.alekhin@gmail.com>

alalek committed:
core: add __NetBSD__ build fix in parallel.cpp

alalek committed:
dnn(ocl4dnn): cleanup dead code, improve logging

alalek committed:
core(tls): avoid destruction of TlsAbstraction singleton

JonasVautherin committed:
Merge pull request #20801 from JonasVautherin:fix-gst-error-handling

* Fix gst error handling

* Use the return value instead of the error, which gives no guarantee of being NULL in case of error
* Test err pointer before accessing it
* Remove unreachable code

* videoio(gstreamer): restore check in writer code

Co-authored-by: Alexander Alekhin <alexander.a.alekhin@gmail.com>

alalek committed:
Merge pull request #20837 from alalek:fixup_20825

alalek committed:
Merge pull request #20838 from alalek:fix_20833

alalek committed:
Merge pull request #20839 from alalek:issue_20828_netbsd

alalek committed:
Merge pull request #20840 from alalek:dnn_ocl_cleanup_code

alalek committed:
Merge pull request #20842 from alalek:valgrind_suppression

alalek committed:
Merge pull request #20841 from alalek:core_keep_TlsAbstraction_singleton_3.4

alalek committed:
Merge remote-tracking branch 'upstream/3.4' into merge-3.4

alalek committed:
release: OpenCV 3.4.16

alalek committed:
Merge tag '3.4.16'

alalek committed:
Merge pull request #20818 from rogday:yolov4x_mish_cuda

fengyuentau committed:
Merge pull request #20422 from fengyuentau:dnn_face

Add DNN-based face detection and face recognition into modules/objdetect

* Add DNN-based face detector impl and interface

* Add a sample for DNN-based face detector

* add recog

* add notes

* move samples from samples/cpp to samples/dnn

* add documentation for dnn_face

* add set/get methods for input size, nms & score threshold and topk

* remove the DNN prefix from the face detector and face recognizer

* remove default values in the constructor of impl

* regenerate priors after setting input size

* two filenames for readnet

* Update face.hpp

* Update face_recognize.cpp

* Update face_match.cpp

* Update face.hpp

* Update face_recognize.cpp

* Update face_match.cpp

* Update face_recognize.cpp

* Update dnn_face.markdown

* Update dnn_face.markdown

* Update face.hpp

* Update dnn_face.markdown

* add regression test for face detection

* remove underscore prefix; fix warnings

* add reference & acknowledgement for face detection

* Update dnn_face.markdown

* Update dnn_face.markdown

* Update ts.hpp

* Update test_face.cpp

* Update face_match.cpp

* fix a compile error for python interface; add python examples for face detection and recognition

* Major changes for Vadim's comments:

* Replace class name FaceDetector with FaceDetectorYN in related failes

* Declare local mat before loop in modules/objdetect/src/face_detect.cpp

* Make input image and save flag optional in samples/dnn/face_detect(.cpp, .py)

* Add camera support in samples/dnn/face_detect(.cpp, .py)

* correct file paths for regression test

* fix convertion warnings; remove extra spaces

* update face_recog

* Update dnn_face.markdown

* Fix warnings and errors for the default CI reports:

* Remove trailing white spaces and extra new lines.

* Fix convertion warnings for windows and iOS.

* Add braces around initialization of subobjects.

* Fix warnings and errors for the default CI systems:

* Add prefix 'FR_' for each value name in enum DisType to solve the
redefinition error for iOS compilation; Modify other code accordingly

* Add bookmark '#tutorial_dnn_face' to solve warnings from doxygen

* Correct documentations to solve warnings from doxygen

* update FaceRecognizerSF

* Fix the error for CI to find ONNX models correctly

* add suffix f to float assignments

* add backend & target options for initializing face recognizer

* add checkeq for checking input size and preset size

* update test and threshold

* changes in response to alalek's comments:

* fix typos in samples/dnn/face_match.py

* import numpy before importing cv2

* add documentation to .setInputSize()

* remove extra include in face_recognize.cpp

* fix some bugs

* Update dnn_face.markdown

* update thresholds; remove useless code

* add time suffix to YuNet filename in test

* objdetect: update test code

alalek committed:
highgui(docs): we don't support 32-bit integer images in imshow()

alalek committed:
js: fix build of SIMD tests

alalek committed:
plugins: reverse candidates order to fetch higher versions first

alalek committed:
Merge pull request #20847 from alalek:fix_build_js_simd_tests

alalek committed:
Merge pull request #20846 from alalek:issue_19947

YanivHollander committed:
Merge pull request #20721 from YanivHollander:DocFixes

Doc fixes

* Update imgproc.hpp

* Update imgproc.hpp

alalek committed:
Merge pull request #20848 from alalek:reverse_plugins_candidates_order

alalek committed:
release: OpenCV 4.5.4

alalek committed:
Merge tag '4.5.4'

rogday committed:
fix const - input and remove unimplemented function

rogday committed:
change asserts for Sum

alalek committed:
Merge pull request #20859 from rogday:sub_const_fix

MarkGHX committed:
Fixed comments

MarkGHX committed:
Address the comments

MarkGHX committed:
Update CMakeLists.txt

alalek committed:
Merge pull request #20860 from rogday:sum_fix

alalek committed:
core: ensure 'int' result from CV_POPCNT_U64(x)

alalek committed:
Merge pull request #20834 from sthalik:msvc-shared-library-static-crt

icestraw committed:
fix type error of slice indices

pkubaj committed:
Add support for runtime CPU feature check on POWER on FreeBSD.

1. Code uses PPC_FEATURE_HAS_VSX, but it's not checked similarly to
PPC_FEATURE2_ARCH_3_00 and PPC_FEATURE2_ARCH_3_00 for availability. FreeBSD has
those macros in machine/cpu.h, but I went with the way chosen for
PPC_FEATURE2_ARCH_3_00 and PPC_FEATURE2_ARCH_3_00. Other than that, FreeBSD also
has sys/auxv.h and that's where elf_aux_info() is defined.
2. getauxval() is actually Linux-only, but code checked for __unix__. It won't
work on all UNIX, so change it back to __linux__. Add another code variant
strictly for FreeBSD.
3. Update comment. This commit adds code for FreeBSD, but recently there
appeared support for powerpc64 in OpenBSD.

alalek committed:
Merge pull request #20867 from alalek:fix_CV_POPCNT_U64_msvs

rogday committed:
add Ceil, Floor, Log, Round, Sqrt, Not, Equal, Less, Greater

alalek committed:
Merge pull request #20868 from icestraw:fix-type-error-of-slice-indices

Wehzie committed:
Fixed typo in CV_Error message

Error was "Input parameters must be a matrices!", but "matrices" is plural and doesn't allow the unspecific article "a".

flytogcp committed:
Update qrcode.cpp

Fixed issue #20880, QRDetect::searchHorizontalLines() boundary condition will skip the matched qrcode near the end

Nicholas-Ho-arm committed:
Merge pull request #20740 from Nicholas-Ho-arm:3.4_SymmColumnVec_32f8u

* Add SymmColumnVec_32f8u

* Fix double to float warnings

alalek committed:
Merge remote-tracking branch 'upstream/3.4' into merge-3.4

alalek committed:
cmake: update installation of python extra submodules

- support Python standalone builds
- loader installs submodules unconditionally

alalek committed:
Merge pull request #20885 from alalek:cmake_fix_python_submodules

alalek committed:
Merge pull request #20879 from Wehzie:patch-1

alalek committed:
dnn(int8): fix using of incorrect UMat constructor

alalek committed:
core(SIMD): update int64 SSE constructor

sergiud committed:
fixed AVX compile error

Some older compilers do not allow to pass a `const int` as an immediate. Use an unnamed enum instead.

sivanov-work committed:
Merge pull request #20773 from sivanov-work:merge_vpl_source_unite

G-API: oneVPL (simplification) unite components in entire VPL source

* Unify components in VPLSource

* Revert back decode WRN & Add compile guard

* Address come comments

* Add source alias

* Apply comment for exception handling

Commit:
Allow to set av_log_set_level to reduce ffmpeg level below AV_LOG_ERROR

TolyaTalamanov committed:
Merge pull request #20856 from TolyaTalamanov:at/cfg-batch-size

G-API: Extend ie::Params to specify batch size

* Add cfgBatchSize to ie::Params

* Fix comments to review

alalek committed:
Merge pull request #20895 from sergiud:avx-error-immediate

alalek committed:
samples: update parallel_backend examples

- use find_package(TBB)

alalek committed:
samples: fix build without threading support

alalek committed:
samples: fix build without threading support

alalek committed:
Merge pull request #20897 from ThalesGroup:master

rogday committed:
Merge pull request #20883 from rogday:eltwise_refactoring

* backport elementwise_layers refactor

* keep NULL

alalek committed:
Merge pull request #20894 from alalek:core_simd_int64_ctor_sse

zchrissirhcz committed:
Merge pull request #20900 from zchrissirhcz:3.4-hwfeatures-support-qnx

* fix: correctly check neon flags for QNX platform

* refactor: change __QNXNTO__ to __QNX__

alalek committed:
Merge pull request #20892 from alalek:issue_20888

alalek committed:
4.x: drop DISABLE_OPENCV_24_COMPATIBILITY macro

not used in 4.x code

Commit:
changed no longer patented SIFT

sivanov-work committed:
Merge pull request #20738 from sivanov-work:merge_master_vpl_dev_select

 G-API: oneVPL - Implement IDeviceSelector & default cfg_param-based selector

* Initial commit

* Add MACRO undef

* Change IDeviceSelector, Change Inf sample for choose external device

* Fix compilation

* Address some comments

* Fix compilation

* Add missing header

* Add EXPORT to dev selector

* Add guard

* Remove enum type attr

* Fix compilation without VPL

* Add HAVE_INFER guard in sample

* Remove unusable include from tests

* Remove unusable include from sample

* Remove cl_d3d11 header from unit test

alalek committed:
Merge pull request #20910 from lukasalexanderweber:sift_fix

MaximMilashchenko committed:
Merge pull request #19721 from MaximMilashchenko:Audio

add audio support in cap_msmf

* audio msmf

* fixed warnings

* minor fix

* fixed SampleTime MSMF

* minor fix, fixed audio test, retrieveAudioFrame

* fixed warnings

* impelemented sync audio and video stream with start offset

* fixed error

* fixed docs

* fixed audio sample

* CAP_PROP_AUDIO_POS, minor fixed

* fixed warnings

* videoio(MSMF): update audio test checks, add debug logging

* fixed

* fixed desynchronization of time positions, warnings

* fixed warnings

* videoio(audio): tune tests checks

* videoio(audio): update properties description

* build warnings

Co-authored-by: Alexander Alekhin <alexander.a.alekhin@gmail.com>

alalek committed:
Merge pull request #20877 from rogday:simple_layers

alalek committed:
Merge pull request #20908 from alalek:4.x_drop_legacy

Commit:
32bit rgb bmp file should not copy data as rgba

alalek committed:
hotfix: repair Clang ABI

berak committed:
resolves #20913

imgproc: remove asserts for circles_ in HoughCircles

alalek committed:
Merge pull request #20920 from alalek:hotfix_clang_abi

alalek committed:
Merge pull request #20916 from Harvey-Huang:32bit_rgb

alalek committed:
Merge pull request #20906 from alalek:backport_20903

alalek committed:
Merge pull request #20903 from alalek:samples_fix_nothreads_build

mshabunin committed:
Merge pull request #20614 from mshabunin:use-onevpl-load

videoio: use oneVPL load mechanism, encoder bitrate estimation

* videoio: updated oneVPL support - use mfxLoad

* videoio: advanced bitrate estimation for MFX encoder

* videoio: improved MediaSDK/oneVPL/libva detection

* videoio(ffmpeg): don't try oneVPL

* videoio(test): tune checks of videoio_mfx.read_write_raw tests

Co-authored-by: Alexander Alekhin <alexander.a.alekhin@gmail.com>

Commit:
fix findMinEnclosingTriangle and add tests

sivanov-work committed:
Merge pull request #20921 from sivanov-work:atl_patch

G-API: FIX OpenVINO build - Add CComPtr RAII replacement into G-API sample

* Add CComPtr RAII replacement into G-API sample

* Remove completely `atlbase.h'

alalek committed:
Merge pull request #20912 from AleksandrPanov:fix_findMinEnclosingTriangle

Commit:
fix gkernel Doxygen documentation

Commit:
Offer more accurate perf test on native

Commit:
Add better perf tests for both native and web

Commit:
Modify per tests for better results

Harvey-Huang committed:
Merge pull request #20875 from Harvey-Huang:master

* bmp specified BI_BITFIELDS should take care RGBA bit mask

* change the name

* support xrgb bmp file

* support xrgb bmp file(add test case)

* update testing code

alalek committed:
Merge pull request #20919 from berak:patch-1

alalek committed:
Merge remote-tracking branch 'upstream/3.4' into merge-3.4

Commit:
Use more latest version of Electron

alalek committed:
Merge pull request #20940 from alalek:cmake_fix_samples_parallel_backend

alalek committed:
cmake: find_package with QUIET

MaximMilashchenko committed:
Merge pull request #20942 from MaximMilashchenko:AudioPatch

Audio patch

* fixed microphone, audio position

* fixed docs

* changed AudioOpenCheck

alalek committed:
Merge pull request #20947 from alalek:cmake_find_package_quiet

Commit:
fixed bug: opencv read tif file convert Palette color image as grayscale image

alalek committed:
Merge pull request #20924 from alexgiving:atrutnev/fix_gkernel_documentation

shengyu7697 committed:
remove redundant semicolons

crywang committed:
Merge pull request #20935 from crywang:dnn_face

Fix problems in tutorial and python sample of dnn_face.

* Update dnn_face.markdown

* Update face_match.py

noahstier committed:
tvl1 cuda optflow optimization

alalek committed:
Merge pull request #20954 from shengyu7697:remove-redundant-semicolon

alalek committed:
Merge pull request #20958 from noahstier:3.4_tvl1_optimization

alalek committed:
Merge pull request #20950 from Harvey-Huang:master

alalek committed:
cmake: fix popcnt detection with Intel Compiler

alalek committed:
build(icc): fix nodiscard attribute handling

alalek committed:
core(simd): fix compilation with MSVC-Clang

alalek committed:
Merge pull request #20960 from alalek:fix_icc_support_3.4

alalek committed:
Merge pull request #20961 from alalek:fix_msvc_clang

Commit:
Merge pull request #20937 from mpashchenkov:mp/ocv-gapi-warnings

G-API: Disable Windows warnings with 4996 code

* Windows warnings 4503 and 4996 are disabled with dnn style

* Applying comments to review

* Reproducing

* Added check MSVC_VERSION for both warnings

alalek committed:
build: fix MSVC-Clang warnings about unused parameters in stubs

alalek committed:
core: eliminate Winvalid-noreturn in base.hpp

s-trinh committed:
Add conventional Bayer naming.

alalek committed:
Merge pull request #20967 from alalek:fix_msvc_clang_unused_parameter

alalek committed:
Merge pull request #20969 from alalek:fix_msvc_clang_warning_noreturn

alalek committed:
cmake: use CMAKE_BUILD_TYPE=Release by default

Commit:
Support latest WebNN Clamp op

Isydmr committed:
Correct drive links for DB_IC15 and DB_TD500

alalek committed:
Merge pull request #20972 from Isydmr:correct-db-text-detector-drive-links

alalek committed:
python: properly handle step for multichannel case

Commit:
Merge pull request #20922 from alexgiving:atrutnev/align_expect_assert_macros

GAPI: Align EXPECT/ASSERT macros

* Align TEST macros

* restart CI

* Fix ASSERT_GT in gapi_async_test

JuliaBareeva committed:
Added overflow handling during conversion from float to int for LinearFilter

alalek committed:
Merge pull request #20971 from alalek:cmake_build_type_use_release

alalek committed:
Merge pull request #20976 from alalek:issue_20968

alalek committed:
highgui: drop invalid cvGetWindowImageRect

- return type is C++ template
- removal from 'extern "C"' scope broke ABI anyway, so this symbols is removed completelly

alalek committed:
highgui: drop invalid cvGetWindowImageRect

alalek committed:
Merge pull request #20980 from alalek:highgui_fix_cvGetWindowImageRect_3.4

alalek committed:
Merge pull request #20982 from alalek:highgui_drop_cvGetWindowImageRect_4.x

alalek committed:
Merge remote-tracking branch 'upstream/3.4' into merge-3.4

Commit:
Merge pull request #20836 from alexgiving:atrutnev/rename_Adapter_to_IAdapter

* Rename RMat::Adapter to RMat::IAdapter

* Add comments

alalek committed:
videoio(dshow): eliminate build warnings from MSVC-Clang

alalek committed:
Merge pull request #20970 from s-trinh:update_Bayer_naming

Commit:
Add definition of HAVE_WEBNN macro

alalek committed:
Merge pull request #20999 from alalek:dnn_replace_deprecated_calls

dnn(protobuf): replace deprecated calls

* dnn: replace deprecated ByteSize() => ByteSizeLong()

* dnn: replace deprecated calls, use GetRepeatedFieldRef

APrigarina committed:
add test case for QR detect fix

alalek committed:
Merge pull request #20990 from alalek:fix_warnings_msvc_clang_dshow_3.4

alalek committed:
Merge pull request #20882 from flytogcp:flytogcp-patch-1

nikpappas committed:
Fix trackbar in falsecolor cpp sample

alalek committed:
Merge pull request #21003 from APrigarina:add_test_qrdetect_fix

alalek committed:
dnn(cmake): don't hijack OpenCL options with Tengine

alalek committed:
Merge pull request #21007 from alalek:cmake_dnn_fix_wrong_tengine_order

alalek committed:
dnn(test): skip tests with high memory usage

- 32-bit configuration may fail due to memory fragmentation

alalek committed:
Merge pull request #21009 from alalek:dnn_skip_tests_32bit

vrabaud committed:
Only use fma functions when CV_FMA3 is set.

In practice, processors offering AVX2/AVX512 also FMA, that is why it got unnoticed.

alalek committed:
Merge pull request #21011 from vrabaud:3.4

alalek committed:
Merge remote-tracking branch 'upstream/3.4' into merge-3.4

alalek committed:
Merge pull request #21005 from nikpappas:bug-samples-falsecolor-trackbar

alalek committed:
highgui(win32): fix trackbar setRange

alalek committed:
Merge pull request #21016 from alalek:issue_21002

lukasalexanderweber committed:
Merge pull request #21020 from lukasalexanderweber:squash

Created Stitching Tool based on stitching_detailed.py

anonymouss committed:
videoio: drop unnecessary offset for accessing video output buffer

Fix: #21021

NDK API AMediaCodec_getOutputBuffer() returns MediaCodecBuffer::data()
which is actually ABuffer::data(). The returned buffer address is already
adjusted by offset.

More info:
    ABuffer::base() returns base address without offset
    ABuffer::data() returns base + offset

Change-Id: I2936339ce4fa9acf657a5a7d92adc1275d7b28a1

alalek committed:
Merge pull request #21027 from anonymouss:4.x

Crayon-new committed:
Merge pull request #20904 from Crayon-new:fix_bug_in_maxLayer

fix bug: wrong output dimension when "keep_dims" is false in pooling layer.

* fix bug in max layer

* code align

* delete permute layer and add test case

* add name assert

* check other cases

* remove c++11 features

* style:add "const" remove assert

* style:sanitize file names

Commit:
fix markers parse in gen_pattern.py

alalek committed:
Merge pull request #20998 from alalek:update_protobuf_3.19.1

3rdparty(protobuf): upgrade 3.5.2 => 3.19.1

* 3rdparty(protobuf): upgrade 3.5.2 => 3.19.1

* dnn: update protobuf files (3.19.1)

* 3rdparty(protobuf): re-apply OpenCV patch for custom fields (3.19.1)

* protobuf: suppress new build warnings

* protobuf: remove unused files

Commit:
Merge pull request #20664 from anna-khakimova:ak/resize_simd

Fluid: SIMD for Resize Linear 8UC3

* SIMD for fluid::Resize 8U3C

* Rework horizontal pass + add 8U4C case

* Reproduce stackoverflow test

* StackOverflow test

* SSE42 impl

* SSE42 impl improvement

* GAPI:SSE42 simd opt for Resize 8UC3. Final version

* Fix tests

* Conditional compilation fix

* Applied comments

* Applied comments. Step2

* Applied comments. Step2

sivanov-work committed:
Merge pull request #20901 from sivanov-work:merge_source_unite_perf_mod

 G-API: oneVPL - Performance: Add async decode pipeline & add cached pool

* Add async decode pipeline & intro cached pool

* Fix performacne test with checking OPENCV_EXTRA

* Add sip perf test with no VPL

* Fix misprint

* Remove empty line..

* Apply some comments

* Apply some comments

* Make perf test fail if no OPENCV_TEST_DATA_PATH declared

alalek committed:
Merge pull request #20870 from pkubaj:master

SegaraRai committed:
use `#pragma message` instead of `#warning` for non-GCC compilers

tv3141 committed:
Merge pull request #21030 from tv3141:fix_seg_fault_houghlinespointset

Fix seg fault houghlinespointset

* Clarify parameter doc for HoughLinesPointSet

* Fix seg fault.

* Add regression test.

* Fix latex typo

Commit:
Merge pull request #21040 from alexgiving:atrutnev/rename_RMatAdapter

GAPI: Rename MatAdapter to RMatOnMat

alalek committed:
Merge pull request #20977 from JulieBar:remap_overflow

alalek committed:
Merge pull request #21028 from AleksandrPanov:fix_markers_parse_in_gen_pattern.py

alalek committed:
Merge pull request #21025 from alalek:issue_21004

* dnn(ocl4dnn): fix LRN layer accuracy problems

- FP16 intermediate computation is not accurate and may provide NaN values

* dnn(test): update tolerance for FP16

Commit:
Support group convolution

ffontaine committed:
modules/videoio/src/cap_ffmpeg_impl.hpp: fix build with gcc 4.8

Fix the following build failure with gcc 4.8:

In file included from /home/buildroot/autobuild/instance-3/output-1/build/opencv4-4.5.4/modules/videoio/src/cap_ffmpeg_impl.hpp:100:0,
                 from /home/buildroot/autobuild/instance-3/output-1/build/opencv4-4.5.4/modules/videoio/src/cap_ffmpeg.cpp:50:
/home/buildroot/autobuild/instance-3/output-1/build/opencv4-4.5.4/modules/videoio/src/cap_ffmpeg_hw.hpp: In constructor 'HWAccelIterator::HWAccelIterator(cv::VideoAccelerationType, bool, AVDictionary*)':
/home/buildroot/autobuild/instance-3/output-1/build/opencv4-4.5.4/modules/videoio/src/cap_ffmpeg_hw.hpp:939:23: error: use of deleted function 'std::basic_istringstream<char>& std::basic_istringstream<char>::operator=(const std::basic_istringstream<char>&)'
             s_stream_ = std::istringstream(accel_list);
                       ^
In file included from /home/buildroot/autobuild/instance-3/output-1/host/opt/ext-toolchain/arm-none-linux-gnueabi/include/c++/4.8.3/complex:45:0,
                 from /home/buildroot/autobuild/instance-3/output-1/build/opencv4-4.5.4/modules/core/include/opencv2/core/cvstd.inl.hpp:47,
                 from /home/buildroot/autobuild/instance-3/output-1/build/opencv4-4.5.4/modules/core/include/opencv2/core.hpp:3306,
                 from /home/buildroot/autobuild/instance-3/output-1/build/opencv4-4.5.4/modules/videoio/include/opencv2/videoio.hpp:46,
                 from /home/buildroot/autobuild/instance-3/output-1/build/opencv4-4.5.4/modules/videoio/src/precomp.hpp:57,
                 from /home/buildroot/autobuild/instance-3/output-1/build/opencv4-4.5.4/modules/videoio/src/cap_ffmpeg.cpp:42:
/home/buildroot/autobuild/instance-3/output-1/host/opt/ext-toolchain/arm-none-linux-gnueabi/include/c++/4.8.3/sstream:272:11: note: 'std::basic_istringstream<char>& std::basic_istringstream<char>::operator=(const std::basic_istringstream<char>&)' is implicitly deleted because the default definition would be ill-formed:
     class basic_istringstream : public basic_istream<_CharT, _Traits>
           ^
/home/buildroot/autobuild/instance-3/output-1/host/opt/ext-toolchain/arm-none-linux-gnueabi/include/c++/4.8.3/sstream:272:11: error: use of deleted function 'std::basic_istream<char>& std::basic_istream<char>::operator=(const std::basic_istream<char>&)'

Fixes:
 - http://autobuild.buildroot.org/results/60f8846b435dafda0ced412d59ffe15bdff0810d

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>

Commit:
Implement Scale_layer using WebNN

alalek committed:
Merge pull request #21048 from ffontaine:4.x

Commit:
Merge pull request #20857 from alexgiving:atrutnev/move_API_samples

Move API sample code to tutorial_code location

alalek committed:
Merge pull request #21039 from SegaraRai:use-pragma-message

alalek committed:
Merge remote-tracking branch 'upstream/3.4' into merge-3.4

Commit:
Add Softmax option for native classification example

Commit:
Merge remote-tracking branch 'upstream/4.x' into gsoc_2021_webnn

Commit:
Fix comments

MarkGHX committed:
Fix comments

Commit:
Fix existing bugs
